### PR TITLE
Ability to load local files in a WebView on iOS devices

### LINKED ISSUE #4443: Can't load local files in a WebView on iOS devices
I hear you think: "not this one again!?". Bear with me.

### Tell us about the problem
Loading a local file like `<WebView src="~/webview-content.html"/>` fails when doing `tns run ios` on an *iOS device*. A simulator is fine, as well as Android.

The problem is the space in the calculated path to the file: `/var/mobile/Containers/Data/Application/C015FB8D-32BA-43FF-A518-DDA6B5A4F5EA/Library/Application Support/LiveSync/app/path-to-webview-content.html`, the space at `Application Support` breaks loading the content. Replacing it with `%20` fixes it.

### Which platform(s) does your issue occur on?
iOS

### Please provide the following version numbers that your issue occurs with:
- CLI: 3.0.3
- Cross-platform modules: 3.0.1
- Runtime(s): tns-ios 3.0.1
- Plugin(s): n/a

### Please tell us how to recreate the issue in as much detail as possible.
Created a [minimal demo app](https://github.com/EddyVerbruggen/nativescript-webview-ios-local-file-issue/tree/master) with instructions.

### Fix
Submitted #4444

### Temporary workaround
Until the PR is merged and released you can load content like this (Angular-TypeScript example, but it the same idea for any stack):

```html
<WebView [src]="webViewSRC"></WebView>
```

```typescript
import * as fs from "tns-core-modules/file-system";

public webViewSRC: string = encodeURI(`${fs.knownFolders.currentApp().path}/path-to-webview-content.html`);
```

Once the issue is fixed you can replace that with:

```typescript
public webViewSRC: string = '~/path-to-webview-content.html';
```

### PR DESCRIPTION
**Pull request by EddyVerbruggen**
Fixes #4443, unittest added.

### CODE REVIEW

File: tests/app/ui/web-view/web-view-tests.ts [hunk 1]
```diff
@@ -87,6 +87,44 @@ export class WebViewTest extends testModule.UITest<webViewModule.WebView> {
         // << webview-localfile
     }
 
+    public testLoadLocalFileWithSpaceInPath(done) {
+        let webView = this.testView;
+
+        // >> webview-localfile
```
Comment (vakrilov) [inline tests/app/ui/web-view/web-view-tests.ts]: Remove this comment

File: tests/app/ui/web-view/web-view-tests.ts [hunk 2]
```diff
@@ -87,6 +87,44 @@ export class WebViewTest extends testModule.UITest<webViewModule.WebView> {
...
+        webView.on(webViewModule.WebView.loadFinishedEvent, function (args: webViewModule.LoadEventData) {
+            // >> (hide)
```
Comment (vakrilov) [inline tests/app/ui/web-view/web-view-tests.ts]: Remove this comment

File: tests/app/ui/web-view/web-view-tests.ts [hunk 3]
```diff
@@ -87,6 +87,44 @@ export class WebViewTest extends testModule.UITest<webViewModule.WebView> {
...
+
+            try {
+                TKUnit.assertNull(args.error, args.error);
+                TKUnit.assertEqual(actual, webView.ios ? expectedHtml : expectedTitle, "File ~/ui/web-view/test.html not loaded properly.");
+                done(null);
+            }
+            catch (e) {
+                done(e);
+            }
+            // << (hide)
```
Comment (vakrilov) [inline tests/app/ui/web-view/web-view-tests.ts]: Remove this comment

File: tests/app/ui/web-view/web-view-tests.ts [hunk 4]
```diff
@@ -87,6 +87,44 @@ export class WebViewTest extends testModule.UITest<webViewModule.WebView> {
...
+            let message;
+            if (!args.error) {
+                message = "WebView finished loading " + args.url;
+            }
+            else {
+                message = "Error loading " + args.url + ": " + args.error;
+            }
+        });
+        webView.src = "~/ui/web-view/test with spaces.html";
+        // << webview-localfile
```
Comment (vakrilov) [inline tests/app/ui/web-view/web-view-tests.ts]: Remove this comment

**Comment by vakrilov** [comment]
10x @EddyVerbruggen!
Can you please the unneeded comments and we can merge this one.

**Comment by EddyVerbruggen** [comment]
Thanks! The comments are gone now, good to learn what those are used for 👍

**Pull request merged**